dawn/src/driver-includes/to_vtk.cpp

```diff
@@ -117,16 +117,17 @@ static std::string formatNaNs(const double value) {
 
 namespace {
 StencilFieldsVtkOutput& getStencilFieldsVtkOutput(int num_k, std::string stencil_name, int iter) {
-  if(stencil_to_output_map.count(std::make_pair(stencil_name, iter - 1))) {
-    stencil_to_output_map.erase(std::make_pair(stencil_name, iter - 1));
-  }
   if(stencil_to_output_map.count(std::make_pair(stencil_name, iter)) == 0) {
     stencil_to_output_map.emplace(std::make_pair(stencil_name, iter),
                                   StencilFieldsVtkOutput(num_k, stencil_name, iter));
   }
   return stencil_to_output_map.at(std::make_pair(stencil_name, iter));
 }
 
+void flushAtIter(std::string stencil_name, int iter) {
+  stencil_to_output_map.erase(std::make_pair(stencil_name, iter - 1));
```

Comment:
... this solution is cleaner, however it adds implicitly an assumption that was not there before: that the same stencil (same stencil name string) won't be called more than once in the same time iteration. Therefore I suggest to add this assumption somewhere here: https://docs.google.com/document/d/1ac0Z_8K_2mxU_fsR-9I_ih0KWry34bPOeydD8KyDiHM/edit#heading=h.1uyg0j7e7p8t
```suggestion
  stencil_to_output_map.erase(std::make_pair(stencil_name, iter));
```
Also, why flushing the previous iteration here? I think you want to flush the current one in `flushAtIter()`